python/observation-publisher/publisher/prediction_log_parser.py

```diff
@@ -20,8 +21,11 @@ class PredictionLogFeatureTable:
     def from_struct(
         cls, table_struct: Struct, inference_schema: InferenceSchema
     ) -> Self:
-        assert isinstance(table_struct["columns"], ListValue)
-        columns = list_value_as_string_list(table_struct["columns"])
+        if inference_schema.feature_orders is not None:
+            columns = inference_schema.feature_orders
+        else:
+            assert isinstance(table_struct["columns"], ListValue)
+            columns = list_value_as_string_list(table_struct["columns"])
```

Comment:
Is there any scenario when both conditions true (inference_schema.feature_orders is not None and table_struct["columns"] is ListValue instance)? And when this happens, inference_schema.feature_orders will take precedence?

Reply:
For users who use PyfuncV3 model, `columns` are expected to be present, and is expected to be in the correct order. Hence, there is no need for the user to explicitly specify the feature orders. Specifying the feature orders is mandatory only for the case of standard model.

So it is possible for both condition to return True, though it shouldn't be in the normal workflow. If it does happen, i believe it's fine for feature_orders to take precedence